bin/compile

```diff
@@ -1,77 +1,105 @@
     export POETRY_VERSION=1.1.4
     log "No Poetry version specified in POETRY_VERSION config var. Defaulting to $POETRY_VERSION."
 else
     log "Using Poetry version from POETRY_VERSION config var: $POETRY_VERSION"
 fi
 
-function indent() {
-        c='s/^/       /'
-        case $(uname) in
-                Darwin) sed -l "$c";;
-                *)      sed -u "$c";;
-        esac
-}
-
 log "Generate requirements.txt with Poetry"
 
 log "Install Poetry"
 
 curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py \
-  | sed -e 's/allowed_executables = \["python", "python3"\]/allowed_executables = ["python3"]/' \
-  | python3 \
-  | indent
+    | sed -e 's/allowed_executables = \["python", "python3"\]/allowed_executables = ["python3"]/' \
+    | python3 \
+    | indent
+
+PATH_TO_POETRY_ENV="$HOME/.poetry/env"
 
-source $HOME/.poetry/env
+if [ -f "$PATH_TO_POETRY_ENV" ] ; then
+    # shellcheck source=/dev/null
+    . "$PATH_TO_POETRY_ENV"
+fi
 
-log "Export requirements.txt from Poetry"
+REQUIREMENTS_FILE="requirements.txt"
+
+log "Export $REQUIREMENTS_FILE from Poetry"
 
 cd "$BUILD_DIR"
 
+EXPORT_PARAMETERS=(--without-hashes --with-credentials)
+
+if [ -n "${POETRY_EXPORT_DEV_REQUIREMENTS:-}" ] && [ "$POETRY_EXPORT_DEV_REQUIREMENTS" != "0" ] ; then
+    log "Enable exporting dev requirements to $REQUIREMENTS_FILE"
+    EXPORT_PARAMETERS+=(--dev)
+fi
+
 # pip can't handle vcs & editable dependencies when requiring hashes (https://github.com/pypa/pip/issues/4995)
 # poetry exports local dependencies as editable by default (https://github.com/python-poetry/poetry/issues/897)
-poetry export -f requirements.txt --without-hashes --with-credentials | sed 's/^-e //' > requirements.txt
+poetry export -f "$REQUIREMENTS_FILE" "${EXPORT_PARAMETERS[@]}" > "$REQUIREMENTS_FILE.orig"
+sed -e 's/^-e //' < "$REQUIREMENTS_FILE.orig" > "$REQUIREMENTS_FILE" && rm "$REQUIREMENTS_FILE.orig"
 
 RUNTIME_FILE="runtime.txt"
 
 log "Export Python version from Poetry to Heroku $RUNTIME_FILE file"
 
-if [ "$DISABLE_POETRY_CREATE_RUNTIME_FILE" ] && [ "$DISABLE_POETRY_CREATE_RUNTIME_FILE" != "0" ]; then
-  log "Skip generation of $RUNTIME_FILE file from poetry.lock"
-  exit 0
+if [ -n "${DISABLE_POETRY_CREATE_RUNTIME_FILE:-}" ] && [ "$DISABLE_POETRY_CREATE_RUNTIME_FILE" != "0" ] ; then
+    log "Skip generation of $RUNTIME_FILE file from poetry.lock"
+    exit 0
 fi
 
-if [ -f "$RUNTIME_FILE" ]; then
-  log "$RUNTIME_FILE found, delete this file from your repository!"
-  exit 1
+if [ -f "$RUNTIME_FILE" ] ; then
+    log "$RUNTIME_FILE found, delete this file from your repository!" >&2
+    exit 1
 fi
 
-# Get Python version from poetry.lock file metadata section
-VERSION=$(sed -n '/^\[metadata\]/,/^\[/p' poetry.lock | sed -n -e 's/^python-versions\s*=\s*//p' | tr -d '\"')
+if [ -z "${PYTHON_RUNTIME_VERSION:-}" ] ; then
+    log "Read Python version from poetry.lock"
+    # Get Python version from poetry.lock file metadata section
```

Comment:
I think that comment isn't necessary as the log entry above already tells so.

Reply:
Well, it was there before the rewrite, but I can remove it, np.